src/web/src/components/SignUp/Forms/GitHubAccount.tsx

```diff
@@ -0,0 +1,172 @@
+/* eslint-disable camelcase */
+import { useEffect, useState } from 'react';
+import { createStyles, makeStyles, Theme } from '@material-ui/core';
+import { connect } from 'formik';
+import useSWR from 'swr';
+
+import { SignUpForm } from '../../../interfaces';
+import formModels from '../Schema/FormModel';
+import { TextInput, CheckBoxInput } from '../FormFields';
+import PostAvatar from '../../Posts/PostAvatar';
+
+const { githubUsername, github: githubModel, githubOwnership } = formModels;
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    root: {
+      padding: '0',
+      margin: '0',
+      width: '100%',
+      position: 'relative',
+      minHeight: '100%',
+    },
+    container: {
+      display: 'grid',
+      gridTemplateAreas: '1fr',
+      textAlign: 'center',
+      justifyItems: 'center',
+      alignItems: 'start',
+      width: '100%',
+      position: 'absolute',
+      minHeight: '100%',
+      [theme.breakpoints.down(600)]: {
+        width: '90%',
+        marginLeft: '5%',
+      },
+    },
+    titlePage: {
+      fontSize: '1.5em',
+    },
+    subtitlePage: {
+      fontSize: '1.1em',
+      lineHeight: '1.8em',
+    },
+    infoContainer: {
+      display: 'grid',
+      gridTemplateColumns: '65% 35%',
+      gridGap: '1%',
+      textAlign: 'center',
+      justifyItems: 'center',
+      alignItems: 'center',
+      width: '90%',
+      [theme.breakpoints.down(600)]: {
+        gridTemplateColumns: '1fr',
+      },
+    },
+    inputsContainer: {
+      width: '100%',
+      display: 'grid',
+      gridTemplateColumns: '100%',
+      '& .MuiFormHelperText-root': {
+        fontSize: '0.9em',
+        color: 'black',
+      },
+      '& .MuiFormLabel-root': {
+        color: 'black',
+      },
+    },
+    avatarPreview: {
+      textAlign: 'center',
+      justifyItems: 'center',
+      alignItems: 'center',
+      justifySelf: 'end',
+      padding: '6%',
+      borderRadius: '5px',
+      [theme.breakpoints.down(600)]: {
+        justifySelf: 'center',
+        padding: '3%',
+        marginTop: '5%',
+      },
+    },
+    username: {
+      fontSize: '1.1em',
+    },
+  })
+);
+
+const gitHubApiUrl = 'https://api.github.com/users';
+
+const GitHubAccount = connect<{}, SignUpForm>((props) => {
+  const classes = useStyles();
+  const { values, setFieldValue } = props.formik;
+  const [username, setUsername] = useState(values.githubUsername);
+  const [inputTimeout, setInputTimeout] = useState(setTimeout(() => {}, 0));
```

Comment:
This is not a great pattern, because the GitHub API will rate-limit you if you make too many requests.  If a user types, deletes, types, etc over and over we may get locked out.  I'd suggest you add a button so they can indicate they are done, and do the request then.

This can happen now or in a follow-up, but make sure you file before you land this.

Reply:
I imagine that would happen only when users type slowly, or they want to abuse the feature by feeding multiple values.
The profile is only fetched `1000ms` after users stop typing, which makes just 1 request when I test it.

Reply:
Regardless, let's use a button.  Network-requests-as-you-type are not a great idea

Reply:
Ok, we'll fix this in this PR.

Reply:
Done.